bin/runtime/src/lib.rs

```diff
@@ -519,13 +518,13 @@ impl_runtime_apis! {
         }
     }
 
-    impl primitives::AlephApi<Block, AlephId, BlockNumber> for Runtime {
-        fn current_session() -> Option<AuthoritySession<AlephId,BlockNumber>> {
-            Aleph::current_session()
-        }
-
-        fn next_session() -> Option<AuthoritySession<AlephId,BlockNumber>> {
-            Aleph::next_session()
+    impl primitives::AlephSessionApi<Block, AlephId, BlockNumber> for Runtime {
+        fn current_session() -> AuthoritySession<AlephId,BlockNumber> {
+            AuthoritySession {
+                session_id: Session::session_index(),
+                authorities: Aleph::authorities(),
+                stop_h: Session::estimate_next_new_session(System::block_number()).0.unwrap(),
```

Comment:
Does it returns start_h of the next session or start_h -1?

Reply:
We to Session pallet `Period` and `Offset` as parameters. This will always return `Offset + k*Period` here. If current block is `n`, then  `Offset + k*Period > n`. So for example if `n = Offset + k*Period` it will return `Offset + (k+1)*Period`